extide/gradle/src/org/netbeans/modules/gradle/loaders/LegacyProjectLoader.java

```diff
@@ -425,10 +439,95 @@ private static NbProjectInfo retrieveProjectInfo(NbGradleProjectImpl projectImpl
         wasOnline.set(true);
-        ret = action.run();
-        return ret;
+        return runInfoAction(action);
+    }
+    
+
+    /**
+     * Makes a workaround for standard {@link PipedOutputStream} wait.
+     * <p>The {@link PipedInputStream#read()}, in case the receive buffer is
+     * empty at the time of the call, waits for up to 1000ms.
+     * {@link PipedOutputStream#write(int)} does call <code>sink.receive</code>,
+     * but does not <code>notify()</code> the sink object so that read's
+     * wait() terminates.
+     * <p>
+     * As a result, the read side of the pipe waits full 1000ms even though data
+     * become available during the wait.
+     * <p>
+     * The workaround is to simply {@link PipedOutputStream#flush} after write,
+     * which returns from wait()s immediately.
+     *
+     * @author Svata Dedic Copyright (C) 2020
+     */
+    static class ImmediatePipedOutputStream extends PipedOutputStream {
+        @Override
+        public void write(byte[] b, int off, int len) throws IOException {
+            super.write(b, off, len);
+            flush();
+        }
+
+        @Override
+        public void write(int b) throws IOException {
+            super.write(b);
+            flush();
+        }
     }
 
+    private static NbProjectInfo runInfoAction(BuildActionExecuter<NbProjectInfo> action) {
+        class LogDelegate implements Runnable {
+            final BufferedReader rdr;
+            
+            LogDelegate(InputStream is) throws IOException {
+                rdr = new BufferedReader(new InputStreamReader(is, "UTF-8"));
+            }
+            
+            public void run() {
+                boolean first = true;
+                try {
+                    String line;
+                    while ((line = rdr.readLine()) != null) {
+                        if (first) {
+                            LOG.log(Level.FINER, "[gradle] ---- daemon log starting");
+                            first = false;
+                        }
+                        LOG.log(Level.FINER, "[gradle] {0}", line);
+                    }
+                } catch (IOException ex) {
+                } finally {
+                    LOG.log(Level.FINER, "[gradle] ---- log terminated");
+                }
+            }
+        }
+        
+        OutputStream logStream = null;
+        try {
+            if (LOG.isLoggable(Level.FINER)) {
+                if (LOG.isLoggable(Level.FINEST)) {
+                    action.addArguments("--debug"); // NOI18N
+                }
+                PipedOutputStream pos = new ImmediatePipedOutputStream();
+                try {
+                    logStream = pos;
+                    DAEMON_LOG_RP.post(new LogDelegate(new PipedInputStream(pos)));
+                } catch (IOException ex) {
+                    throw new IllegalStateException(ex);
+                }
+                action.setStandardOutput(pos);
```

Comment:
Update: No, gradle output from the `LegacyProjectLoader` BuildAction goes to /dev/null unless explicitly not redirected somewhere. 
The Piped stuff was the simplest that I could think of as Gradle outputs byte stream and I need to detect newlines among the (potential) UTF8 multibyte garbage + decode Strings from the stream. It could be probably done directly in a special OutputStream implementation. Or maybe I could rely on `flush()` to be called after each line (?)